test/e2e/storage/non_csi_volumes.go

```diff
@@ -0,0 +1,709 @@
+	volSource v1.VolumeSource
+	testFile  string
+	podSec    v1.PodSecurityContext
+	fileSizes []int64
+}
+
+type inlineVolumeTest struct {
+	name    string
+	config  framework.VolumeTestConfig
+	fsGroup *int64
+	tests   []framework.VolumeTest
+}
+
+type persistentVolumeModeTest struct {
+	name     string
+	config   framework.VolumeTestConfig
+	pvSource v1.PersistentVolumeSource
+
+	scName           string
+	volBindMode      storagev1.VolumeBindingMode
+	volMode          v1.PersistentVolumeMode
+	isBlockSupported bool
+}
+
+type testDriver interface {
+	createDriver()
+	cleanupDriver()
+	createInlineVolumeTest() inlineVolumeTest
+	createInlineVolumeIOTest() inlineVolumeIOTest
+	createPersistentVolumeModeTest() []persistentVolumeModeTest
+}
+
+var testDrivers = map[string]func(f *framework.Framework, config framework.VolumeTestConfig) testDriver{
+	"NFS":                        initNFSDriver,
+	"GlusterFS":                  initGlusterFSDriver,
+	"iSCSI [Feature:Volumes]":    initISCSIDriver,
+	"Ceph-RBD [Feature:Volumes]": initRbdDriver,
+	"CephFS [Feature:Volumes]":   initCephFSDriver,
+}
+
+var _ = utils.SIGDescribe("Non-CSI", func() {
+	f := framework.NewDefaultFramework("volumes")
+
+	var (
+		cs     clientset.Interface
+		ns     *v1.Namespace
+		config framework.VolumeTestConfig
+	)
+
+	BeforeEach(func() {
+		cs = f.ClientSet
+		ns = f.Namespace
+		config = framework.VolumeTestConfig{
+			Namespace: ns.Name,
+			Prefix:    "volume",
+		}
+	})
+
+	for driverName, initDriver := range testDrivers {
+		curDriverName := driverName
+		curInitDriver := initDriver
+
+		Context(fmt.Sprintf("using driver: %s", curDriverName), func() {
+			var (
+				driver testDriver
+			)
+
+			BeforeEach(func() {
+				driver = curInitDriver(f, config)
+				driver.createDriver()
+			})
+
+			AfterEach(func() {
+				driver.cleanupDriver()
+			})
+
+			// Tests originally executed in volumes.go
+			It("should be mountable", func() {
+				t := driver.createInlineVolumeTest()
+				defer framework.VolumeTestCleanup(f, t.config)
+
+				framework.TestVolumeClient(cs, t.config, t.fsGroup, t.tests)
+			})
+
+			// Tests originally executed in volume_io.go
+			It("should write files of various sizes, verify size, validate content [Slow]", func() {
+				t := driver.createInlineVolumeIOTest()
+				err := testVolumeIO(f, cs, t.config, t.volSource, &t.podSec, t.testFile, t.fileSizes)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			// Tests originally executed in persistent_volumes-volumemode.go
+			It("should show a volume with proper volume mode to a pod", func() {
+				tests := driver.createPersistentVolumeModeTest()
+				for _, t := range tests {
+					testVolumeMode(f, cs, t.config, ns.Name, t.scName, t.volBindMode, t.volMode, t.pvSource, t.isBlockSupported)
+				}
+			})
+		})
+	}
+})
+
+////////////////////////////////////////////////////////////////////////
+// NFS
+////////////////////////////////////////////////////////////////////////
```

Comment:
not sure we have comment blocks like that in the repo, but if sig-storage are fine with that. :) :+1:

Reply:
Thank you for your comments. Actually, they are coming from existing codes. I'm ready to change it, if required.